### DISCUSSION TITLE
map hjlk to scroll-px by default to avert disaster

### DISCUSSION THREAD

**Discussion by p0gster** [qutebrowser/qutebrowser#7897] [General] [answered]
Sometimes when pressing hjkl to scroll a website using the scroll command, the arrow keys are sent to the website instead of scrolling the page. I changed this to scroll-px after reading https://github.com/qutebrowser/qutebrowser/discussions/7754 .
The problem with this default is that in filling out forms, this can cause options to change while the user is attempting to scroll. This could potentially have disastrous consequences for the user IRL by choosing an incorrect option in an important form. I think they should be mapped to scroll-px by default to avoid this possibility even if emulating keypresses with scroll has slightly better behaviour.

**Comment by mschilli87** [comment]
I think that if a user fills an important form and doesn't double-check before submitting, any disaster following is on them. If you press down in Chrome or Safari, it scrolls down. If you press down while focussing on a form field, it will change change the form selection. So by default qutebrowser behaves like every other major browser and I don't see the problem. I actually like being able to interact with forms using vim keys. So your suggestion would break my experience. I wouldn't mind because I can change the setting but so can you. Thus I think the default should mirror what most browsers do, which is the case currently already.

**Reply by p0gster** [reply]
I could go and double check the form, go down by pressing j, and ctrl-d. The j would change the option in the form and the ctrl-d would take me to the bottom of the form. So even if I double check my form, this could cause disaster.

**Reply by mschilli87** [reply]
Then tab through every form that your life depends on and submit it via return on the submit button. 😉 
Still, I fail to see how this has anything to do with qutebrowser. Maybe whatever process we talk about should not rely on a manually submitted web form as the only layer of protection from chaos and carnage. 🤷

**Reply by p0gster** [reply]
I use tab and shift-tab to go through forms mostly. On drop down menus, you don't want jk to be arrow keys as you can normally type the first few letters of the item and go to it. jk mapped to arrow keys is an inconvenience as in normal mode, You should to be able to move around a little without expecting side effects, e.g. changing random fields in the form which is very much unexpected behaviour without being forced to tab around by default to account for this unexpected behaviour.

**Reply by The-Compiler** [reply, marked as answer]
In addition to what's been said already, things have been this way [for over 8 years](https://github.com/qutebrowser/qutebrowser/commit/f49dba6e38a5ed4fbd6a2d8e9920310c2c22ed74), and I believe this is the first time this ever came up.

Perhaps things might change once we have a better way to control scrolling (#3691). Until that's the case, I do feel like this is a very isolated issue and does not justify changing the defaults.
